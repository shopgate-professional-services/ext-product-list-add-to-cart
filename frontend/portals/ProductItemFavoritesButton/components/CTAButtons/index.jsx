import React from 'react';
import PropTypes from 'prop-types';
import FavoritesButton from '@shopgate/pwa-ui-shared/FavoritesButton';
import AddToCartPicker from './components/AddToCartPicker/index';
import { ADD_TO_CART_BUTTON_TYPE_DEFAULT } from '../../../../constants';
import connect from './connector';
import styles from './style';

/**
 * Renders CTA buttons for product page (add to cart + toggle favorites).
 *
 * @param {Object} props Props.
 * @returns {JSX}
 * @constructor
 */
const CTAButtons = (props) => {
  const addToCartButtonProps = {
    buttonSize: styles.cartButtonSize,
    iconSize: styles.cartButtonIconSize,
    type: ADD_TO_CART_BUTTON_TYPE_DEFAULT,
  };
  return (
    <div className={styles.buttons}>
      <FavoritesButton
        active={props.isFavorite}
        productId={props.productId}
        className={styles.favButton}
        rippleClassName={styles.ripple}
      />
      <AddToCartPicker
        buttonProps={addToCartButtonProps}
      />
    </div>
  );
};

CTAButtons.propTypes = {
  isFavorite: PropTypes.bool.isRequired,
  productId: PropTypes.string.isRequired,
};

export default connect(CTAButtons);
