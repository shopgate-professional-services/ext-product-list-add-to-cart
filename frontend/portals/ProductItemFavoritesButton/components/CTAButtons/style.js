import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { variables } = themeConfig;

// Cart ratio is approx 2.33/1
const cartButtonSize = (variables.big * 2.33);
const cartButtonIconSize = variables.big;

const buttons = css({
  position: 'absolute',
  right: variables.gap.small,
  transform: 'translate3d(0, -60%, 0)',
  display: 'flex',
  justifyContent: 'flex-end',
  alignItems: 'center',
  '> :last-child': {
    marginLeft: variables.gap.small,
  },
}).toString();

const favButton = css({
  zIndex: 1, // Prevents the icons to be visible outside of the circle
  fontSize: 12,
}).toString();

const ripple = css({
  padding: 8,
}).toString();

export default {
  buttons,
  cartButtonSize,
  cartButtonIconSize,
  favButton,
  ripple,
};
