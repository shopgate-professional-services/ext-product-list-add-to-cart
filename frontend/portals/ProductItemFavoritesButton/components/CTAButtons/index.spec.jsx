import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore } from '@shopgate/pwa-common/store';
import { Provider } from 'react-redux';
import CTAButtons from './index';

jest.mock('@shopgate/pwa-ui-shared/FavoritesButton', () => <div>FavoritesButton</div>);
jest.mock('./components/AddToCartPicker/index', () => <div>AddToCartPicker</div>);

jest.mock('../../../../selectors', () => ({
  isProductOnFavoriteList: true,
}));

const store = createMockStore();

describe('CTAButtons', () => {
  it('should render CTAButtons', () => {
    const component = shallow((
      <Provider store={store}>
        <CTAButtons />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
  });
});
