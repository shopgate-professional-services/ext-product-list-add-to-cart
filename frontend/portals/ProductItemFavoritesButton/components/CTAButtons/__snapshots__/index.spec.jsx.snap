// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAButtons should render CTAButtons 1`] = `<Connect(CTAButtons) />`;
